### DISCUSSION TITLE
Run multiple commands on a query

### DISCUSSION THREAD

**Discussion by phirestalker** [beetbox/beets#4497] [Q&A] [answered]
Is it possible to run multiple commands on my library and only write files when they are all finished?

I have a plugin that I created, which will create a par2 file for each track. To be useful it must recreate the par2 after each file change. As you can imagine this makes running multiple commands on my library take quite a while. I would like to run a set of commands (say lastgenre, acousticbrainz, and lyrics) on my entire library, but only write files when they are all finished to only create the par2 files once per file.

Also, I want the files to have the tags written to the files and the database, so I do not have to rely on any program (beets included) to query, play, or create playlists from my collection.

**Comment by sampsyo** [comment, marked as answer]
I think the easiest way to do this would be to temporarily disable the `import.write` config option, do your work, and then run the `beet write` command. Counterintuitively, the `import.write` option affects *many* commands—not just the import process.

**Reply by sampsyo** [reply]
That's the idea, but I would give it a try first!

**Reply by phirestalker** [reply]
Sorry for the last comment, I forgot how to read.
Is there a commandline option to disable write so I don't have to edit the config file?
i.e, beet -w no acousticbrainz

**Reply by sampsyo** [reply]
There's nothing exactly like that, but you could consider using the `-C` option to load a (very short) configuration file that just overrides that one setting.
